### PR TITLE
ONYX-11120 - Audience variable integration into main jwt flow

### PR DESCRIPTION
**Pull request by sashaCher**
### What does this PR do?

Integrates `audience` variable into main `authn-jwt` flow.
Automation sanity tests.

### What ticket does this PR close?
Resolves #[relevant GitHub issues, eg 76]

#### Change log
- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [X] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [X] The changes in this PR do not affect the Conjur API

### CODE REVIEW

**Review by tzheleznyak — APPROVED** [review]

File: cucumber/authenticators_jwt/features/authn_jwt_audience.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
+Feature: JWT Authenticator - Check audience
+
+  Verify the authenticator works correctly audience variable and aud claim
+
+  Background:
```
Comment (tzheleznyak) [inline cucumber/authenticators_jwt/features/authn_jwt_audience.feature:5]: Can these test be moved to authn_jwt_check_standard_claims.feature?

File: cucumber/authenticators_jwt/features/authn_jwt_audience.feature [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is <http_code>
+    And The following appears in the log after my savepoint:
+    """
+    <log>
+    """
+    Examples:
+      | Test       | audience        | aud                                          | http_code | log                                                                                                                                       |
```
Comment (tzheleznyak) [inline cucumber/authenticators_jwt/features/authn_jwt_audience.feature:65]: Can you move the `,` to the token instead of the inline

Comment (sashaCher) [reply cucumber/authenticators_jwt/features/authn_jwt_audience.feature:65]: Done

File: cucumber/authenticators_jwt/features/authn_jwt_audience.feature (hunk 1 above)
Comment (sashaCher) [reply cucumber/authenticators_jwt/features/authn_jwt_audience.feature:5]: Agreed

**Review by tzheleznyak — APPROVED** [review]
LGTM

**Review by tzheleznyak — APPROVED** [review]
LGTM

**Pull request merged**